presto/eval.py

```diff
@@ -130,15 +130,19 @@ def finetune_sklearn_model(
             targets = targets.ravel()
 
         fit_models = []
+        class_weights = cast(WorldCerealLabelledDataset, dl.dataset).class_weights
         model_dict = {
             "Regression": LogisticRegression(
-                class_weight="balanced", max_iter=1000, random_state=self.seed
+                class_weight={idx: weight for idx, weight in enumerate(class_weights)},
+                max_iter=1000,
+                random_state=self.seed,
             ),
             "Random Forest": RandomForestClassifier(
-                class_weight="balanced", random_state=self.seed
+                class_weight={idx: weight for idx, weight in enumerate(class_weights)},
+                random_state=self.seed,
             ),
             "CatBoostClassifier": CatBoostClassifier(
-                random_state=self.seed, auto_class_weights="Balanced"
+                random_state=self.seed, class_weights=class_weights
```

Comment:
For consistency with the other methods and just to be sure, I suggest to use the same dictionary here as well, see [documentation](https://catboost.ai/en/docs/references/training-parameters/common#class_weights):

> If class labels are not standard consecutive integers [0, 1 ... class_count-1], use the dict or collections.OrderedDict type with label to weight mapping.